unit_tests/kernels/UnitTestKernelUtils.h

```diff
@@ -172,23 +172,23 @@ void calc_projected_nodal_gradient(
   GenericFieldType& tensorField);
 
 void expect_all_near(
-  const Kokkos::View<double*>& calcValue,
+  const Kokkos::View<double*,Kokkos::LayoutRight,sierra::nalu::MemSpace>& calcValue,
```

Comment:
I never noticed it until now, but `MemSpace` is not a good name to indicate `HostMemSpace`, should we rename this to be consistent with `DeviceMemSpace`, `DeviceShmem`, `HostShmem` etc.?

Reply:
Yes I'm confused about that memspace argument to expect_all_near. Renaming it would probably help. I probably don't have that correct in the TestLinearSystem views. I'm getting a compile error where on gpu I have to comment out one of the expect_all_near functions to avoid a 'ambiguous' error.

Reply:
Is it the templated one that's coming up ambiguous?

Reply:
Here is the compiler output:
`
/scratch/william/source/nalu-wind-alanw0/unit_tests/kernels/UnitTestScalarAdvDiffElem.C(73): error: more than one instance of overloaded function "unit_test_kernel_utils::expect_all_near" matches the argument list:
            function "unit_test_kernel_utils::expect_all_near(const Kokkos::View<double *> &, const double *, double)"
            function "unit_test_kernel_utils::expect_all_near(const Kokkos::View<double **> &, const double *, double)"
            argument types are: (Kokkos::View<double *, Kokkos::Cuda::array_layout, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::Cuda::memory_space>>, const double [8]
`
Strangely, it seems to be confusing the 1D and 2D kokkos views... It gives the same message with and without the MemSpace template parameter.

Reply:
In KokkosInterface.h it looks like we have MemSpace defined as CudaUVMSpace so that's almost certainly not what we want...

Reply:
Should the argument to `expect_all_near` be of type

```
Kokkos::View<double*,Kokkos::LayoutRight, sierra::nalu::MemSpace>::HostMirror
Kokkos::View<double**,Kokkos::LayoutRight, sierra::nalu::MemSpace>::HostMirror
```

Reply:
@alanw0 How are you interpreting this 

```
Kokkos::View<double *, Kokkos::Cuda::array_layout,Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::Cuda::memory_space>>
```

It seems to me that execution is HostSpace, but memory is CudaSpace?

Reply:
Yes I noticed that, it doesn't seem right. I wonder if it's related to MemSpace == CudaUVMSpace in KokkosInterface.h. Do you think we should set MemSpace to just be Kokkos::CudaSpace?

Reply:
@alanw0 the `Kokkos::View` you are sending to the function is the hostview right? you are not comparing the gold values on device. So the memory space shouldn't be CUDA at all?

Reply:
Yes, sending the hostview, specifically the 'RHSView::HostMirror'. I'm going to try making MemSpace be CudaSpace, and then HostMirror should presumably have nothing to do with cuda. I'll let you know what that does.

Reply:
No combination I could think worked. I'm kind of stuck on this. I've resorted to renaming one of the expect_all_near functions to expect_all_near_2d. I'll push another commit to this pull-request and see what you think.

Reply:
Since it is just a unit test utility, seems a minor thing to have both expect_all_near and expect_all_near_2d.

Reply:
I agree with @overfelt, renaming the function will clarify things.